src/Bicep.LangServer/BicepParamsCompilationManager.cs

```diff
@@ -50,24 +50,24 @@ public void UpsertCompilation(DocumentUri uri, int? version, string text, string
         {
 
             var paramsFile = SourceFileFactory.CreateBicepParamFile(uri.ToUri(), text);
+            var semanticModel = new ParamsSemanticModel(paramsFile);
 
             var usingDeclarations = paramsFile.ProgramSyntax.Children.OfType<UsingDeclarationSyntax>();
             var bicepFilePath = PathHelper.TryGetUsingPath(usingDeclarations.SingleOrDefault(), out var getUsingPathFailureBuilder);
 
-            if(bicepFilePath == null){
-                throw new Exception("Path to Bicep file not provided");
-            }
-
-            if(!Uri.TryCreate(uri.ToUri(), bicepFilePath, out var bicepFileUri)){
-                throw new Exception("Can not create Uri for linked Bicep file");
+            if(bicepFilePath == null)
```

Comment:
This is the part where I read the corresponding template file in LS. This behavior is different in the CLI because there I throw an error if we try to build with no template linked. I was wondering if there are cleaner ways to do this? Maybe move the file reading logic in a helper method?
https://github.com/Azure/bicep/blob/57c2d2207bfca63a42e7daa0715e717f19355703/src/Bicep.Cli/Services/CompilationService.cs#L116

Reply:
Did you mean `!=` null here?

I think the checks could be simplified to:

```csharp
if (PathHelper.TryGetUsingPath(usingDeclarations.SingleOrDefault(), out var bicepFilePath, out var _) 
    && Uri.TryCreate(uri.ToUri(), bicepFilePath, out var bicepFileUri))
```

Reply:
I see, that makes sense